common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java

```diff
@@ -69,15 +69,13 @@ public Cause diagnoseCorruption(
       String host,
       int port,
       String execId,
-      int shuffleId,
-      long mapId,
-      int reduceId,
+      String blockId,
       long checksum,
       String algorithm) {
     try {
       TransportClient client = clientFactory.createClient(host, port);
       ByteBuffer response = client.sendRpcSync(
-        new DiagnoseCorruption(appId, execId, shuffleId, mapId, reduceId, checksum, algorithm)
+        new DiagnoseCorruption(appId, execId, blockId, checksum, algorithm)
```

Comment:
Why are we going from explicit types to an opaque string ?
This will make it more unmaintainable as code evolves.

Reply:
It gives us the flexiblility to extend these block operations to other block types.